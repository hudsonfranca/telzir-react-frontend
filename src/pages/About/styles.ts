import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  min-height: 100vh;
`;

export const SectionOne = styled.section`
  width: 100%;
  height: 100vh;
  position: relative;
  z-index: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: var(--white);

  h1 {
    font-size: 4rem;
    padding-bottom: 20px;
    width: 80%;
    text-align: center;
  }

  p {
    font-size: 2rem;
    width: 80%;
    text-align: center;
  }
  &::after {
    content: '';
    width: 100%;
    height: 100%;
    background: rgba(11, 12, 16, 0.5);
    position: absolute;
    top: 0;

    z-index: -1;
  }
`;

export const Section = styled.div<{ bgColor: string }>`
  width: 100%;
  min-height: 100vh;
  background-color: ${({ bgColor }) => bgColor};
  padding: 30px;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-template-rows: repeat(auto-fit, minmax(250px, 1fr));
  grid-gap: 30px;
  justify-content: space-between;
  align-items: center;

  div {
    width: 100%;
    height: 100%;
    color: var(--white);
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    h1 {
      font-size: 5rem;
      margin-bottom: 50px;
      text-align: left;
    }

    h2 {
      font-size: 2rem;
      text-align: left;
    }
  }
`;

export const Image = styled.div<{ bgUrl: string }>`
  width: 100%;
  height: 100%;
  position: relative;
  z-index: 1;
  background-image: url(${({ bgUrl }) => bgUrl});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;

  &::after {
    content: '';
    width: 100%;
    height: 100%;
    background: rgba(11, 12, 16, 0.6);
    position: absolute;
    top: 0;
    z-index: 0;
  }
`;
