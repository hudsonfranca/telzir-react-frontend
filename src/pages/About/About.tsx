import React from 'react';

import { Parallax } from 'react-parallax';
import { Container, SectionOne, Section, Image } from './styles';
import about from '../../assets/about.jpg';
import company1 from '../../assets/company1.jpg';
import company2 from '../../assets/company2.jpg';

const About: React.FC = () => {
  document.title = 'Sobre nós';
  return (
    <Container data-test="aboutContainer">
      <Parallax
        bgImage={about}
        strength={500}
        bgClassName="react-parallax-bgimage"
      >
        <SectionOne data-test="aboutSectionOne">
          <h1> Saiba tudo sobre nós.</h1>

          <p>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi
            exercitationem provident facilis nulla nam vero quod est magnam
            dolorum itaque ipsum, eligendi quam labore optio explicabo? Ad
            molestiae laboriosam quam?
          </p>
        </SectionOne>
      </Parallax>
      <Section bgColor="var(--secondary)" data-test="aboutSectionTwo">
        <div data-test="aboutContentSectionTwo">
          <h1>Lorem ipsum dolor sit amet consectetur</h1>
          <h2>
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Tempore,
            accusantium assumenda. Obcaecati vel itaque voluptatibus. Lorem
            ipsum dolor sit amet, consectetur adipisicing elit. Ipsam,
            consequuntur?
          </h2>
        </div>
        <Image bgUrl={company1} data-test="imageSectionTwo" />
      </Section>

      <Section bgColor="var(--primary)" data-test="aboutSectionThree">
        <Image bgUrl={company2} data-test="imageSectionThree" />
        <div data-test="aboutContentSectionThree">
          <h1>Lorem ipsum dolor sit amet consectetur</h1>
          <h2>
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Tempore,
            accusantium assumenda. Obcaecati vel itaque voluptatibus. Lorem
            ipsum dolor sit amet, consectetur adipisicing elit. Ipsam,
            consequuntur?
          </h2>
        </div>
      </Section>
    </Container>
  );
};

export default About;
