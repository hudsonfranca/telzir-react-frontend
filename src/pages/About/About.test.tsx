import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import About from './About';
import 'jest-styled-components';

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<About />);
});

describe('<About /> redering', () => {
  it('Should render without errors', () => {
    const about = renderer.create(<About />).toJSON();

    expect(about).toMatchSnapshot();
  });

  it('Should render 1 aboutContainer', () => {
    expect(wrapper.find(`[data-test='aboutContainer']`).length).toEqual(1);
  });

  it('Should render 1 aboutSectionOne', () => {
    expect(wrapper.find(`[data-test='aboutSectionOne']`).length).toEqual(1);
  });

  it('Should render 1 aboutSectionTwo', () => {
    expect(wrapper.find(`[data-test='aboutSectionTwo']`).length).toEqual(1);
  });

  it('Should render 1 aboutSectionThree', () => {
    expect(wrapper.find(`[data-test='aboutSectionThree']`).length).toEqual(1);
  });

  it('Should render 1 imageSectionTwo', () => {
    expect(wrapper.find(`[data-test='imageSectionTwo']`).length).toEqual(1);
  });

  it('Should render 1 imageSectionThree', () => {
    expect(wrapper.find(`[data-test='imageSectionThree']`).length).toEqual(1);
  });

  it('Should render 1 aboutContentSectionTwo', () => {
    expect(wrapper.find(`[data-test='aboutContentSectionTwo']`).length).toEqual(
      1
    );
  });

  it('Should render 1 aboutContentSectionThree', () => {
    expect(
      wrapper.find(`[data-test='aboutContentSectionThree']`).length
    ).toEqual(1);
  });
});
