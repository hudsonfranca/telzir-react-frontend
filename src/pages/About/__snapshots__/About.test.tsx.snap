// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> redering Should render without errors 1`] = `
.c0 {
  width: 100%;
  min-height: 100vh;
}

.c1 {
  width: 100%;
  height: 100vh;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--white);
}

.c1 h1 {
  font-size: 4rem;
  padding-bottom: 20px;
  width: 80%;
  text-align: center;
}

.c1 p {
  font-size: 2rem;
  width: 80%;
  text-align: center;
}

.c1::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.5);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c2 {
  width: 100%;
  min-height: 100vh;
  background-color: var(--secondary);
  padding: 30px;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(250px,1fr));
  grid-template-rows: repeat(auto-fit,minmax(250px,1fr));
  grid-gap: 30px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 div {
  width: 100%;
  height: 100%;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 div h1 {
  font-size: 5rem;
  margin-bottom: 50px;
  text-align: left;
}

.c2 div h2 {
  font-size: 2rem;
  text-align: left;
}

.c4 {
  width: 100%;
  min-height: 100vh;
  background-color: var(--primary);
  padding: 30px;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(250px,1fr));
  grid-template-rows: repeat(auto-fit,minmax(250px,1fr));
  grid-gap: 30px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 div {
  width: 100%;
  height: 100%;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 div h1 {
  font-size: 5rem;
  margin-bottom: 50px;
  text-align: left;
}

.c4 div h2 {
  font-size: 2rem;
  text-align: left;
}

.c3 {
  width: 100%;
  height: 100%;
  position: relative;
  z-index: 1;
  background-image: url(company1.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c3::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.6);
  position: absolute;
  top: 0;
  z-index: 0;
}

.c5 {
  width: 100%;
  height: 100%;
  position: relative;
  z-index: 1;
  background-image: url(company2.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c5::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.6);
  position: absolute;
  top: 0;
  z-index: 0;
}

<div
  className="c0"
  data-test="aboutContainer"
>
  <div
    className="react-parallax "
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <img
      alt=""
      className="react-parallax-bgimage"
      src="about.jpg"
      style={
        Object {
          "MozBackfaceVisibility": "hidden",
          "MsBackfaceVisibility": "hidden",
          "WebkitBackfaceVisibility": "hidden",
          "WebkitTransform": "translate3d(-50%, 0, 0)",
          "WebkitTransformStyle": "preserve-3d",
          "left": "50%",
          "position": "absolute",
          "transform": "translate3d(-50%, 0, 0)",
        }
      }
    />
    <div
      className="react-parallax-content"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <section
        className="c1"
        data-test="aboutSectionOne"
      >
        <h1>
           Saiba tudo sobre n√≥s.
        </h1>
        <p>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi exercitationem provident facilis nulla nam vero quod est magnam dolorum itaque ipsum, eligendi quam labore optio explicabo? Ad molestiae laboriosam quam?
        </p>
      </section>
    </div>
  </div>
  <div
    className="c2"
    data-test="aboutSectionTwo"
  >
    <div
      data-test="aboutContentSectionTwo"
    >
      <h1>
        Lorem ipsum dolor sit amet consectetur
      </h1>
      <h2>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Tempore, accusantium assumenda. Obcaecati vel itaque voluptatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsam, consequuntur?
      </h2>
    </div>
    <div
      className="c3"
      data-test="imageSectionTwo"
    />
  </div>
  <div
    className="c4"
    data-test="aboutSectionThree"
  >
    <div
      className="c5"
      data-test="imageSectionThree"
    />
    <div
      data-test="aboutContentSectionThree"
    >
      <h1>
        Lorem ipsum dolor sit amet consectetur
      </h1>
      <h2>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Tempore, accusantium assumenda. Obcaecati vel itaque voluptatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsam, consequuntur?
      </h2>
    </div>
  </div>
</div>
`;
